### ISSUE TITLE
Cannot access protected property 'EqualityContract' here

### ISSUE THREAD

**Issue by kristofferjalen** [mattwhitfield/Unitverse#105]
Given:
```csharp
internal static class FooFactory
{
    public static Foo Create(Bar bar) => new();
}
record Foo;
record Bar;
```
Generated test:

```csharp
[TestMethod]
public void CreatePerformsMapping()
{
    // Arrange
    var bar = new Bar();
            
    // Act
    var result = FooFactory.Create(bar);
          
    // Assert
    Assert.AreSame(bar.EqualityContract, result.EqualityContract);
}
```
Gives me this error:
```
Cannot access protected property 'EqualityContract' here.
```

**Comment by mattwhitfield** [comment]
Resolved in 0.127 🎈

**Comment by mattwhitfield** [comment]
Oh - forgot to say - thanks for the issue 😁
